code/modules/food_and_drinks/kitchen_machinery/food_replicator.dm

```diff
@@ -0,0 +1,631 @@
+		return 0
+	operating = 1
+	if(!silent)
+		playsound(loc, 'sound/misc/interference.ogg', 50, 1)
+	var/obj/item/weapon/reagent_containers/food/snacks/food_poly/poly = null
+	switch(sides)
+		if(1)
+			poly = new /obj/item/weapon/reagent_containers/food/snacks/food_poly()
+		if(4)
+			poly = new /obj/item/weapon/reagent_containers/food/snacks/food_poly/tetrahedron()
+		if(6)
+			poly = new /obj/item/weapon/reagent_containers/food/snacks/food_poly/cube()
+		if(8)
+			poly = new /obj/item/weapon/reagent_containers/food/snacks/food_poly/octahedron()
+		if(12)
+			poly = new /obj/item/weapon/reagent_containers/food/snacks/food_poly/dodecahedron()
+		if(20)
+			poly = new /obj/item/weapon/reagent_containers/food/snacks/food_poly/icosahedron()
+	//food_polys are always considered duplicated since they are only made by the replicator, so we don't need to set the duped var since it is set in the food_poly define
+	//food_polys don't replicate reagents since they aren't being replicated from something else.
+	handle_emag_effect(poly)
+	sleep(20)
+	adjust_paste(-sides)
+	if(!silent)		//won't use power when we spit out excess (since that runs silently)
+		use_power(active_power_usage)
+	poly.forceMove(get_turf(src))
+	operating = 0
+	if(!silent)
+		playsound(loc, 'sound/machines/ding.ogg', 50, 1)
+	return 1
+
+//adjust_paste is used to alter and sanity check the food_paste value.
+//if food_paste is above the max_food_paste, it will begin printing off a pile of food polys until it is at or below max again.
+/obj/machinery/food_replicator/proc/adjust_paste(amount = 0)
+	food_paste = max(0, (food_paste + amount))
+
+	if(food_paste > max_food_paste)
+		var/paste_to_waste = food_paste - max_food_paste
+		if(paste_to_waste >= 20)
+			print_poly(20, 1)
+		else if(paste_to_waste >= 12)
+			print_poly(12, 1)
+		else if(paste_to_waste >= 8)
+			print_poly(8, 1)
+		else if(paste_to_waste >= 6)
+			print_poly(6, 1)
+		else if(paste_to_waste >= 4)
+			print_poly(4, 1)
+		else
+			print_poly(1, 1)
+
+/obj/machinery/food_replicator/attack_ai(mob/user)
+	return attack_hand(user)
+
+/obj/machinery/food_replicator/attack_hand(mob/user)
+	if(stat & (BROKEN|NOPOWER))
+		return
+	user.set_machine(src)
+	ui_interact(user)
+
+/obj/machinery/food_replicator/interact(mob/user)
+	if(panel_open || !anchored)
+		return
+	//todo
+
+/obj/machinery/food_replicator/attackby(obj/item/O, mob/user, params)
+	if(operating)
+		return
+	if(default_deconstruction_screwdriver(user, open_icon, off_icon, O))
+		return
+	if(default_unfasten_wrench(user, O))
+		return
+	if(exchange_parts(user, O))
+		return
+	default_deconstruction_crowbar(O)
+
+	if(istype(O, /obj/item/weapon/reagent_containers/food/snacks))
+		if(loaded)
+			to_chat(user, "<span class='warning'>There is already a dish loaded. Recycle or eject it before adding a new dish.</span>")
+			return
+		var/obj/item/weapon/reagent_containers/food/snacks/to_load = O
```

Comment:
At some point I was doing something that did require it (namely, checking for the duped var), and then changed how/where that was done and forgot to remove this typing.